jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -3184,28 +3184,38 @@ void java_bytecode_convert_methodt::save_stack_entries(
 {
   for(auto &stack_entry : stack)
   {
+    // remove typecasts if existing
+    auto entry = std::ref(stack_entry);
+    while(entry.get().id() == ID_typecast)
+    {
+      entry = std::ref(to_typecast_expr(stack_entry).op());
+    }
+
     // variables or static fields and symbol -> save symbols with same id
-    if((write_type==bytecode_write_typet::VARIABLE ||
-        write_type==bytecode_write_typet::STATIC_FIELD) &&
-       stack_entry.id()==ID_symbol)
+    if(
+      (write_type == bytecode_write_typet::VARIABLE ||
+       write_type == bytecode_write_typet::STATIC_FIELD) &&
+      entry.get().id() == ID_symbol)
```

Comment:
It would be nice if the formatting cleanup could be factored out into a separate commit.

Reply:
unfortunately, `clang-format` seems to work on the whole condition, I applied it only on the second conjunct, same for the other cases

Reply:
Oh, I'm sorry, I missed the `stack_entry` -> `entry` change.